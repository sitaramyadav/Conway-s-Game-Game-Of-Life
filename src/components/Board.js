import React from "react";
import styled from "styled-components";
import { Square } from "./Square";
import { ROWS, COLOMS } from "../constants";

const BoardStyle = styled.main`
  padding: 5px 5px;
  height: 900px;
  margin: auto;
  width: 900px;
  box-sizing: border-box;
  border: 5px solid blue;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const Board = ({ boardStatus }) => {
  const renderRow = (tr = []) => {
    if (boardStatus && boardStatus[0]) {
      for (let row = 0; row < ROWS; row++) {
        const td = [];
        for (let colom = 0; colom < COLOMS; colom++) {
          td.push(
            <Square key={`${row},${colom}`} isAlive={boardStatus[row][colom]} />
          );
        }
        tr.push(<tr key={row}>{td}</tr>);
      }
    }
    return (
      <table>
        <tbody>{tr}</tbody>
      </table>
    );
  };

  return <BoardStyle data-testid={"Board"}>{renderRow()}</BoardStyle>;
};
