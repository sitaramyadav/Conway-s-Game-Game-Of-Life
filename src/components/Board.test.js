import React from "react";
import { Board } from "./Board";
import { render } from "@testing-library/react";

describe("Board", () => {
  it("should render the grid", () => {
    const { queryByTestId } = render(<Board />);
    expect(queryByTestId("Board")).toBeInTheDocument();
    expect(queryByTestId).toMatchSnapshot();
  });
});
