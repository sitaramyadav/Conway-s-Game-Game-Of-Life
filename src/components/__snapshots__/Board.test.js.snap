// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should render the grid 1`] = `[Function]`;
